yjit/src/codegen.rs

```diff
@@ -3635,7 +3753,7 @@ fn jit_thread_s_current(
     _cme: *const rb_callable_method_entry_t,
     _block: Option<IseqPtr>,
     _argc: i32,
-    _known_recv_class: *const VALUE,
+    _known_recv_object: *const VALUE,
```

Comment:
It seems weird that you're passing a pointer here.

This should probably just be called `comptime_recv` and be of type `VALUE`, rather than a pointer to a `VALUE` ?

Reply:
Currently there's a caller that passes a null pointer through, and did so when it was a class as well. Would an Option<VALUE> be better, do you think?

Reply:
If you mean the case in invokesuper, `comptime_recv` seems to be available there too.

Reply:
That's the one I mean. I wasn't sure if there was some reason we were passing null instead. Okay, I'll change that to always pass a VALUE.

Reply:
Okay. Should be fixed now.